### PR TITLE
Dynamodb emulator

### PR DESCRIPTION
**Pull request by lightsofapollo**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
@cbaron Hey Chris, please take a look... The emulator code is a bit tricky so reach out if you have questions. I very carefully wrote that bit to avoid port binding contention issues.

File: packages/appsync-emulator-serverless/__test__/server.test.js [hunk 1]
```diff
@@ -107,7 +123,7 @@ describe('subscriptionServer', () => {
       operationName: 'test',
       query: `
       subscription quoteRequest {
-        subscribeToPutQuoteRequest {
+          subscribeToPutQuoteRequest {
```
Comment (cbaron) [inline packages/appsync-emulator-serverless/__test__/server.test.js]: nit: indentation

File: packages/dynamodb-emulator/index.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+  // to possible race conditions between when we find an open
+  // port and bind to it. This situation is particularly common
+  // in jest where each test file is running in it's own process
+  // each competing for the open port.
+  try {
+    const waiter = wait(opts.startTimeout);
+    await Promise.race([
+      new Promise(accept => {
+        function readStdoutBuffer(buffer) {
+          if (buffer.toString().indexOf(opts.port)) {
```
Comment (cbaron) [inline packages/dynamodb-emulator/index.js]: Do you want to:

`if (buffer.toString().indexOf(opts.port) !== -1` ?

Comment (lightsofapollo) [reply packages/dynamodb-emulator/index.js]: ! now I am confused why this even works

**Pull request merged**